pkg/plugins/backendplugin/provider/provider.go

```diff
@@ -65,6 +68,11 @@ var SecretsManagerProvider PluginBackendProvider = func(_ context.Context, p *pl
 	)
 }
 
-var DefaultProvider PluginBackendProvider = func(_ context.Context, p *plugins.Plugin) backendplugin.PluginFactoryFunc {
-	return grpcplugin.NewBackendPlugin(p.ID, p.ExecutablePath())
+func DefaultProviderWithFeatures(features featuremgmt.FeatureToggles) PluginBackendProvider {
+	return func(_ context.Context, p *plugins.Plugin) backendplugin.PluginFactoryFunc {
+		skipEnvVars := features.IsEnabled(featuremgmt.FlagPluginsSkipHostEnvVars)
+		return grpcplugin.NewBackendPlugin(p.ID, p.ExecutablePath(), skipEnvVars)
+	}
 }
+
+var DefaultProvider = DefaultProviderWithFeatures(featuremgmt.WithFeatures())
```

Comment:
I think I would rather move the `features` parameter to this `DefaultProvider` so people don't mistakenly calls that one rather than `DefaultProviderWithFeatures`